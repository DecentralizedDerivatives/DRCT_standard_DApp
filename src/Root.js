import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from './reducers';

const middlewares = [thunk];
let composeEnhancer = compose;

if (process.env.NODE_ENV === 'development') {
  middlewares.push(createLogger());
  composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
}

const Root = ({ children, initialState = {} }) => {
  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancer(applyMiddleware(...middlewares))
  );

  return <Provider store={store}>{children}</Provider>;
};

export default Root;
