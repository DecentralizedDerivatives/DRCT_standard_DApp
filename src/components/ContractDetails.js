import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import FactoryProvider from '../factoryProvider';
import { formatter } from '../formatter';

export class ContractDetails extends Component {
  formatMoney (val, empty) {
    if (!val) { return <span> {empty || '$0'} </span> }
    var cls = val < 0 ? 'warning' : ''
    return <span className={cls}>{formatter.toDollars(val)}</span>
  }
  formatPercent (val, empty) {
    if (!val) { return <span> {empty || '$0'} </span> }
    var cls = val < 0 ? 'warning' : ''
    return <span className={cls}>{formatter.toPercent(val)}</span>
  }
  renderdiv() {
    const network_id = FactoryProvider.getNetworkId();
    const networks = require('../networkProvider');
    const url = networks[network_id].url + '/address/';
    const div = (
        <div style={{margin: '20px', padding: '10px'}}>
          <h2>Factory Contract</h2>
          <div className='detail-segment'>
            <div className='title'>Address</div>
            <div className='detail'>
              <a href={url + this.props.contract.contractAddress}
                target="_blank"
                rel="noopener noreferrer">{this.props.contract.contractAddress}
              </a>
            </div>
          </div>
          <div className='detail-segment'>
            <div className='title'>Duration</div>
            <div className='detail'>{this.props.contract.contractDuration}</div>
          </div>
          <div className='detail-segment'>
            <div className='title'>Multiplier</div>
            <div className='detail'>{this.props.contract.contractMultiplier}</div>
          </div>
          <div className='detail-segment'>
            <div className='title'>Oracle Address</div>
            <div className='detail'>
              <a
                href={url + this.props.contract.oracleAddress}
                target="_blank"
                rel="noopener noreferrer">{this.props.contract.oracleAddress}
              </a>
            </div>
          </div>
          {this.props.tokenAddress && this.props.tokenAddress.length ?
            <div className='detail-segment'>
              <div className='title'>Token Address</div>
              <div className='detail'>
                <a
                  href={url + this.props.tokenAddress}
                  target="_blank"
                  rel="noopener noreferrer">{this.props.tokenAddress}
                </a>
              </div>
            </div>
          : ''}
          {this.props.contract.contractStartPrice > 0 ?
            <div className='detail-segment'>
              <div className='title'>Start Price</div>
              <div className='detail'>{this.formatMoney(this.props.contract.contractStartPrice)}</div>
            </div>
          : ''}
          {this.props.contract.contractGain ?
            <div className='detail-segment'>
              <div className='title'>Current Gain/Loss</div>
              <div className='detail'>{this.formatPercent(this.props.contract.contractGain)}</div>
            </div>
          : ''}
        </div>
      );

    return div;
  }

  render() {
    return (
      <div id="contract-details">
        <div className="modal-background" onClick={this.props.close}></div>
        <div className="modal">
          {this.renderdiv()}
          <div>
            <div style={{width: '100%', textAlign: 'center', margin: '8px auto'}}>
              <button
                type="submit"
                className="btn btn-primary btn-md"
                onClick={this.props.close}>Ok</button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

ContractDetails.propTypes = {
  contract: PropTypes.object.isRequired,
  tokenAddress: PropTypes.string
};

const mapStateToProps = state => ({
  contract: state.contract,
  tokenAddress: state.selected.selectedTokenAddress
});

export default connect(mapStateToProps, { })(ContractDetails);
