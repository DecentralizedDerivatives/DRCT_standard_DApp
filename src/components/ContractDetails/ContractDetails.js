import React, { Component } from 'react';
import PropTypes from 'prop-types';
import withStyles from 'material-ui/styles/withStyles';
import TextField from 'material-ui/TextField';
import Typography from 'material-ui/Typography';
import Dialog, { DialogContent } from 'material-ui/Dialog';
import styles from './styles';
import Dropdown from '../Dropdown';
import { Factory } from '../../ethereum';

class ContractDetails extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    open: PropTypes.bool.isRequired,
    toggle: PropTypes.func.isRequired,
  };
  constructor(){
    super();
    this.state = {
      open: false,
      currency: '',
      amount: 0.1,
      selectedDate: new Date(),
    };
  }

  static durations = ['One weeks', 'Two weeks'];
  static currency = ['BTC/USD', 'ETH/USD'];

  render() {
    const { classes } = this.props;

    return (
      <div>
        <Dialog
          open={this.props.open}
          onClose={this.props.toggle}
          PaperProps={{ className: classes.paper }}
        >
          <DialogContent className={classes.dialogContent}>

            <div className={classes.inputContainer}>
              <Typography className={classes.title}>Factory Contract</Typography>
            </div>
            <div className={classes.inputContainer}>
              <Typography className={classes.title}>Address: {this.props.contractAddress}</Typography>
            </div>
            <div className={classes.inputContainer}>
              <Typography className={classes.title}>Duration: {this.props.contractDuration}</Typography>
            </div>
            <div className={classes.inputContainer}>
              <Typography className={classes.title}>Multiplier: {this.props.contractMultiplier}</Typography>
            </div>
            <div className={classes.inputContainer}>
              <Typography className={classes.title}>Oracle Address: {this.props.oracleAddress}</Typography>
            </div>
          </DialogContent>
        </Dialog>
      </div>
    );
  }
}

export default withStyles(styles)(ContractDetails);
