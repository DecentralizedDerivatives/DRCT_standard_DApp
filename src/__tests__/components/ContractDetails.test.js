import React from 'react';
import { shallow } from 'enzyme';
import { initStore } from '../../Root';

import ContractDetails from '../../components/ContractDetails';

describe('<ContractDetails />', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const wrapper = shallow(
        <ContractDetails store={initStore()} />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
