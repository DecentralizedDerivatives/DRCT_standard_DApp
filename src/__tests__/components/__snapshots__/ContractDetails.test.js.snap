// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContractDetails /> render() renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(ContractDetails)
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "contract": Object {
        "contractAddress": "",
        "contractCurrentPrice": 0,
        "contractDuration": 0,
        "contractGain": 0,
        "contractMultiplier": 0,
        "contractOpenDates": Object {},
        "contractStartPrice": 0,
        "newContract": Object {
          "address": "",
          "amount": 0,
          "currency": "",
          "duration": "",
          "funded": false,
          "fundedTx": "",
          "id": "",
          "startDate": "",
        },
        "newContractCreateError": null,
        "newContractFundsError": null,
        "oracleAddress": "",
        "orderbook": Array [],
        "recentTrades": Array [],
      },
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "contract": Object {
          "contractAddress": "",
          "contractCurrentPrice": 0,
          "contractDuration": 0,
          "contractGain": 0,
          "contractMultiplier": 0,
          "contractOpenDates": Object {},
          "contractStartPrice": 0,
          "newContract": Object {
            "address": "",
            "amount": 0,
            "currency": "",
            "duration": "",
            "funded": false,
            "fundedTx": "",
            "id": "",
            "startDate": "",
          },
          "newContractCreateError": null,
          "newContractFundsError": null,
          "oracleAddress": "",
          "orderbook": Array [],
          "recentTrades": Array [],
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
